modules/simple_bucket/variables.tf

```diff
@@ -85,3 +85,22 @@ variable "encryption" {
   })
   default = null
 }
+
+variable "lifecycle_rules" {
+  description = "The bucket's Lifecycle Rules configuration."
+  type = list(object({
+    # Object with keys:
+    # - type - The type of the action of this Lifecycle Rule. Supported values: Delete and SetStorageClass.
+    # - storage_class - (Required if action type is SetStorageClass) The target Storage Class of objects affected by this Lifecycle Rule.
+    action = any
```

Comment:
nit: Are we switching to any to keep the end user configs more concise? 
vs main module types
https://github.com/terraform-google-modules/terraform-google-cloud-storage/blob/a0782d700d7f2b5c726bf3a42d8c9dc723795a7a/variables.tf#L132

Reply:
This is a temporary type and is due to the fact that terraform does not support optional keys in objects.

In the main module, I am guessing you always need to set `storage_class` on the action to empty or null, even if you only want the type?

Reply:
Note: also can't use map(string) without breaking the interface like how the main module forces matches_storage_class to be a string instead of list of strings like the underlying API. I believe any is better than map(string) because it lets you preserve the underlying API.

Reply:
yeah I agree. `any` lgtm, I was wondering if making a similar change to `any` on the main module is also worthwhile, since it seems to be the best option as we wait for optional keys in obj.

Reply:
+1 to backporting this to the main module.